src/Features/Core/Portable/TodoComments/ITodoCommentService.cs

```diff
@@ -39,10 +27,35 @@ public TodoComment(TodoCommentDescriptor descriptor, string message, int positio
             Message = message;
             Position = position;
         }
+
+        internal TodoCommentInfo Convert(
+            Document document, SourceText text, SyntaxTree tree)
+        {
+            // make sure given position is within valid text range.
+            var textSpan = new TextSpan(Math.Min(text.Length, Math.Max(0, this.Position)), 0);
+
+            var location = tree.GetLocation(textSpan);
+            // var location = tree == null ? Location.Create(document.FilePath, textSpan, text.Lines.GetLinePositionSpan(textSpan)) : tree.GetLocation(textSpan);
+            var originalLineInfo = location.GetLineSpan();
+            var mappedLineInfo = location.GetMappedLineSpan();
+
+            return new TodoCommentInfo
```

Comment:
>TodoCommentInfo [](start = 23, length = 15)

A better naming pattern just occurred to me that we can use for all these serializable structs. For diagnostics, the equivalent is `DiagnosticData`. How about calling these structs `-Data`? E.g. this one could be `TodoCommentData`.

Reply:
sure.  i'll update all of these.